src/Game/AI/AI/aiActorWaterDepthSelect.cpp

```diff
@@ -1,17 +1,57 @@
 #include "Game/AI/AI/aiActorWaterDepthSelect.h"
+#include "KingSystem/ActorSystem/actActor.h"
 
 namespace uking::ai {
 
 ActorWaterDepthSelect::ActorWaterDepthSelect(const InitArg& arg) : ksys::act::ai::Ai(arg) {}
 
 ActorWaterDepthSelect::~ActorWaterDepthSelect() = default;
 
+bool ActorWaterDepthSelect::isChangeable() const {
+    return getCurrentChild()->isChangeable();
+}
+
+bool ActorWaterDepthSelect::isFailed() const {
+    return getCurrentChild()->isFailed();
+}
+
+bool ActorWaterDepthSelect::isFinished() const {
+    return getCurrentChild()->isFinished();
+}
+
 bool ActorWaterDepthSelect::init_(sead::Heap* heap) {
-    return ksys::act::ai::Ai::init_(heap);
+    return true;
 }
 
 void ActorWaterDepthSelect::enter_(ksys::act::ai::InlineParamPack* params) {
-    ksys::act::ai::Ai::enter_(params);
+    if (isDeep() && isUnderwater()) {
+        changeChild("深瀬", params);
+        return;
+    }
+    changeChild("浅瀬", params);
+}
+
+void ActorWaterDepthSelect::calc_() {
+    /*
+        Only if mOneEnterOnly is disabled, change the actor's depth
+    */
+    if (*mOnEnterOnly_s)
+        return;
+
+    auto is_deep{isDeep() && isUnderwater()};
```

Comment:
```suggestion
    auto is_deep = isDeep() && isUnderwater();
```
nit: think this would look more natural